You can't use ClasspathConfigurationSource without an environment within a JAR

We package up a Java application into a fat JAR that contains all the application source code and the dependencies. This allows the application to be run as `java -jar myapp.jar`, however when doing this if I attempt to load a `ClasspathConfigurationSource` without having an environment set it blows up with:

    Caused by: org.cfg4j.source.context.environment.MissingEnvironmentException: Missing environment: Directory doesn't exist: 
	    at org.cfg4j.source.classpath.ClasspathConfigurationSource.getConfiguration(ClasspathConfigurationSource.java:109)
    	at org.cfg4j.source.compose.MergeConfigurationSource.getConfiguration(MergeConfigurationSource.java:62)
	    at org.cfg4j.provider.SimpleConfigurationProvider.getProperty(SimpleConfigurationProvider.java:91)
	    at org.cfg4j.provider.SimpleConfigurationProvider.getProperty(SimpleConfigurationProvider.java:64)

When running unpacked (from IDE) it all works correctly. Specifying and environment also fixes the problem, but this affects other configurations sources (`EnvironmentVariablesConfigurationSource`) and so I'd rather keep the environment empty.

--- a/cfg4j-core/src/main/java/org/cfg4j/source/classpath/ClasspathConfigurationSource.java
+++ b/cfg4j-core/src/main/java/org/cfg4j/source/classpath/ClasspathConfigurationSource.java
@@ -105,7 +105,7 @@ public class ClasspathConfigurationSource implements ConfigurationSource {
     Path pathPrefix = Paths.get(environment.getName());
 
     URL url = getClass().getClassLoader().getResource(pathPrefix.toString());
-    if (url == null) {
+    if (url == null && !environment.getName().isEmpty()) {
       throw new MissingEnvironmentException("Directory doesn't exist: " + environment.getName());
     }
 

